tools/postinstall.ts

```diff
@@ -1,9 +1,9 @@
 import * as execa from 'execa';
 
 /**
- * In certain circumstances we want to skip the below the steps and it may not always
- * be possible to use --ignore-scripts (e.g. if another tool is what is invoking the
- * install command, such as when nx migrate runs). We therefore use and env var for this.
+ * In certain circumstances we might want to skip the below the steps when another
+ * tool is invoking the install command (such as when nx migrate runs).
+ * We therefore use an env var for this.
  */
```

Comment:
I ran `yarn dedupe` and it generated `21 insertions(+), 564 deletions(-)`. Maybe we can add `yarn dedupe` here too?

Reply:
We can't do it in the postinstall I don't think otherwise we end up in a loop when there are changes resulting from the dedupe.

I have pushed a dedicated commit with an alternative approach to ensuring that the lock file is maximally deduped over time: https://github.com/typescript-eslint/typescript-eslint/pull/6162/commits/18657c0f61619f1ef408ccd4c1ae095d55a239e1

Reply:
we can probably setup an action that runs weekly or something to generate a PR?
actually can we add it to the renovate config so it does it automatically? most of our dep upgrades come via dependabot anyways

Reply:
> we can probably setup an action that runs weekly or something to generate a PR?

Hmm yeah that's probably going to be a lot less disruptive to contributors. Will be annoying to put together a PR and have it fail if you didn't prune recently... I'll revert that change and we can do a prune workflow in a follow up

> most of our dep upgrades come via dependabot anyways

@bradzacher isn't it the opposite? We switched from dependabot to renovate, right?

Reply:
yes, that's what I meant! I totally said both on purpose to cover my bases

Reply:
(PS I've removed the call to `yarn dedupe` from ci.yml in this PR now so that we don't cause issues for users unnecessarily per the above comment)

Reply:
See how forward thinking i am? I think so far ahead I forget how far ahead I think.

![bigbraintime](https://user-images.githubusercontent.com/7462525/207472349-b5d8e773-c34b-4228-884e-58a883c3c8bd.jpg)